### PR TITLE
Implemented reports model and mutation

### PR DESCRIPTION
**Pull request by JoshD94**
## Overview

Implementation for the new user reporting feature was added.
Created Report table and added mutations.
Implemented a change in class_scraper.

## Changes Made

Added new Report model with attributes as follows:
        - id: The ID of the report.
        - issue: The issue reported by the user (enumerated options).
        - description: The description of the report/user input
        - created_at: The date and time the report was created in DateTime format
        - user_id: The ID of the user who created the report.
        - gym_id: The ID of the gym associated with the report.

The report model will store 2 manually inputted fields from the user: issue and description, both in the form of a string.
The enumeration for the issue is based on the high-fidelity figma scheme and is as follows:
INACCURATE_EQUIPMENT = 0
INCORRECT_HOURS = 1
INACCURATE_DESCRIPTION = 2
WAIT_TIMES_NOT_UPDATED = 3
OTHER = 4
Currently, the created_at field, user_id, and gym_id are inputted manually from the front end.

Added new CreateReport mutation in the schema with sample input
{
  "createdAt": "2024-10-09T12:00:00Z",
  "description": "no dumbbells",
  "gymId": 1792236,
  "issue": "INACCURATE_EQUIPMENT ",
  "userId": 1
}

The change made in class_scraper is because the syntax for throwing an exception was incorrect. This issue was identified when attempting to set up the project from scratch, meaning it went undetected on already set up projects because the specific branch is never visited.

## Test Coverage

Manually tested on graphql playground.
Testing log:
- Tested normal input: no exceptions raised
- Tested no description: no exceptions raised
- Tested long description: no exceptions raised
- Tested no date: raised exception
- Tested incorrect gym_id: raised exception (violates foreign key constraint)
- Tested no issue: raised exception (invalid input value for enum)
- Tested incorrect user_id: raised exception (violates foreign key constraint)
- Tested incorrect DateTime format: raised exception (expected type DateTime )

### CODE REVIEW

File: src/models/report.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from sqlalchemy.orm import relationship
+from src.database import Base
+import enum
+
+class ReportType(enum.Enum):
+    inaccurate_equipment = 0
+    incorrect_hours = 1
+    inaccurate_description = 2
+    wait_times_not_updated = 3
+    other = 4
```
Comment (Aayush-Agnihotri) [inline src/models/report.py:11]: Convention is usually that enums are all capitalized

File: src/scrapers/class_scraper.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def fetch_classes(num_pages):
             try:
                 gym_class = db_session.query(Class).filter(Class.name == class_name).first()
                 if gym_class is None:
-                    Exception()
+                    raise Exception()
```
Comment (Aayush-Agnihotri) [inline src/scrapers/class_scraper.py:69]: Maybe add a message in the exception to help with debugging if that ever happens

**Review by Aayush-Agnihotri — APPROVED** [review]
LGTM

File: src/models/report.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    id = Column(Integer, primary_key=True)
+    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
+    issue = Column(Enum(ReportType), nullable=False)  # Discrete options (enumerate)
+    description = Column(String, nullable=False)  # Text input
+    created_at = Column(DateTime, nullable=False)  # Timestamp for user submission
+    gym_id = Column(Integer, ForeignKey("gym.id"), nullable=False)  # One to many relationship with gym
+    # Make relationship with gym and user
+    gym = relationship("Gym", back_populates="reports")
+    user = relationship("User", back_populates="reports")
```
Comment (sophiestrausberg) [inline src/models/report.py:36]: On Uplift we sort columns alphabetically with the exception of `id`.

File: src/schema.py [hunk 4]
```diff
@@ -288,11 +305,29 @@ def mutate(self, info, name):
...
+
+    report = graphene.Field(Report)
+
+    def mutate(self, info, description, user_id, issue, created_at, gym_id):
+        report = ReportModel(description=description, user_id=user_id, issue=issue,
+                             created_at=created_at, gym_id=gym_id)
+        db_session.add(report)
+        db_session.commit()
+        return CreateReport(report=report)
+
```
Comment (sophiestrausberg) [inline src/schema.py:324]: We should raise a GraphQLError in the case that user id or gym id does not exist, or if the issue string is an invalid enum value.

**Review by sophiestrausberg — COMMENTED** [review]
Looks good!

**Comment by JoshD94** [comment]
@Aayush-Agnihotri @sophiestrausberg Changes based on comments have been made and pushed.

**Review by sophiestrausberg — APPROVED** [review]

**Pull request merged**